app/javascript/components/ingredients/base-form-component.vue

```diff
@@ -89,6 +121,32 @@ export default {
     units: { type: Array, required: true },
   },
 
+  data() {
+    return {
+      form: {
+        providerId: null,
+        name: '',
+        sku: null,
+        price: '',
+        currency: 'CLP',
+        quantity: '',
+        measure: '',
+      },
+    };
+  },
+
+  created() {
+    this.form = {
+      providerId: this.ingredient.provider_id,
+      name: this.ingredient.name,
+      sku: this.ingredient.sku,
+      price: this.ingredient.price,
+      currency: this.ingredient.currency,
+      quantity: this.ingredient.quantity,
+      measure: this.ingredient.measure,
+    };
```

Comment:
También con destructiring podria quedar más simple en mi opinión

```js
const {
  provider_id: providerId,
  name,
  sku,
  price,
  currency,
  quantity,
  measure,
} = this.ingredient;

this.form = {
  providerId,
  name,
  sku,
  price,
  currency,
  quantity,
  measure,
};
```

El `provider_id: providerId` no debería ser necesario: debería haber un [transformer](https://github.com/axios/axios#request-config) o [interceptor](https://css-tricks.com/stay-dry-using-axios-for-api-requests/#interceptors) que se encargue de transformar las keys del request de `snake_case` a `camelCase`, pues en ruby se ocupa la primera y en JS la segunda, y viceversa cuando estás mandando algo hacia ruby, transformar de `camelCase` a `snake_case`.